Assets/UXF/Scripts/UI/ParticipantListSelection.cs

```diff
@@ -247,18 +247,16 @@ public string Finish()
 
             // update row in table
             foreach (var keyValuePair in completedForm)
-                row[keyValuePair.Key] = keyValuePair.Value;     
+                UpdateDatapoint(row, keyValuePair.Key, keyValuePair.Value);    
 
             // write pplist
             CommitCSV();
             return ppid;
 
         }
 
-        public void UpdateDatapoint(string ppid, string datapointName, object value)
```

Comment:
This is needed because some tasks (e.g. https://github.com/immersivecognition/posture-assessment-vr) use this method externally to update the form with new information from the task

Reply:
If needed, make another method with the same name but different arguments

Reply:
I had not realised that the method was intended to be used publicly. I searched for its use in the rest of the code, and when I couldn't find it being used in any other place, I thought this was your original intention.

Reply:
Yes this part of the code is messy and not well documented

Reply:
If it is being accessed publicly, then I can think of another problem, when the search for ppid does not a yield only a single result (a null or duplicate entries). Then ppList.AsEnumerable().Single(..) will lead to `InvalidOperationException: Operation is not valid due to the current state of the object`. Instead of assuming valid ppid, maybe we can have an exception for that? This could be similar to how duplicate participant IDs are detected in SetPPList.

Reply:
This could be discussed in a separate project issue if that's more apt.

Reply:
An ideal case would be to have unit tests for some of these issues, but your suggestion makes sense, an exception with a readable error message would be helpful.